### PR TITLE
New Process Modularization System

### LINKED ISSUE #107: Allow for Execution of Different Versions of the Same Process
In order to gracefully migrate from one process version to the next, we need to be able to execute different versions of the same process inside a single BPE instance. Since processes use Java classes to execute service-tasks we might need to load different versions of the same class inside the jvm. Some kind of ClassLoader trickery and manual registration as spring beans via BeanDefinitionBuilder might be necessary.

### PR DESCRIPTION
**Pull request by hhund**
Implements a new process modularization system, allowing for execution of multiple versions of the same process. Processes now manage there own FHIR resources and are managed by the BPE with five different states: NEW (not deployed yet), EXCLUDED (previously deployed - jar deleted or excluded by property), RETIRED (retired by property), DRAFT (active SNAPSHOT builds), ACTIVE (active production).

New process module system allows for external implementation of processes without modifying the DSF code base. Automatically handles management of related FHIR meta data resources.

Adds new :not modifier for code based Search-Parameters

closes #107

### CODE REVIEW

**Review by wetret — APPROVED** [review]
Nice work! I added some comments.

I think there is a `plugin` folder with a README.md in `dsf-bpe-server-jetty` and a `processes` folder with a README.md in `dsf-bpe-server-jetty > docker`  missing.

File: dsf-bpe/dsf-bpe-process-ping/src/test/java/org/highmed/dsf/bpe/start/Ping3MedicFromTtpExampleStarter.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public static void main(String[] args)
 
 		FhirContext context = FhirContext.forR4();
 		ReferenceCleaner referenceCleaner = new ReferenceCleanerImpl(new ReferenceExtractorImpl());
-		FhirWebserviceClient client = new FhirWebserviceClientJersey("https://ttp/fhir/", trustStore, keyStore,
+		FhirWebserviceClient client = new FhirWebserviceClientJersey("https://localhost:8001/fhir/", trustStore, keyStore,
```
[the remark this answers is not in the document]
Comment (hhund) [reply dsf-bpe/dsf-bpe-process-ping/src/test/java/org/highmed/dsf/bpe/start/Ping3MedicFromTtpExampleStarter.java]: fixed in 035048e

File: dsf-bpe/dsf-bpe-process-ping/src/test/java/org/highmed/dsf/bpe/start/Ping3MedicFromTtpExampleStarter.java [hunk 2]
```diff
@@ -45,9 +45,9 @@ public static void main(String[] args)
 		task.setIntent(TaskIntent.ORDER);
 		task.setAuthoredOn(new Date());
 		task.getRequester().setType("Organization").getIdentifier()
-				.setSystem("http://highmed.org/fhir/NamingSystem/organization-identifier").setValue("Test_TTP");
+				.setSystem("http://highmed.org/fhir/NamingSystem/organization-identifier").setValue("Test_Organization");
 		task.getRestriction().addRecipient().setType("Organization").getIdentifier()
-				.setSystem("http://highmed.org/fhir/NamingSystem/organization-identifier").setValue("Test_TTP");
+				.setSystem("http://highmed.org/fhir/NamingSystem/organization-identifier").setValue("Test_Organization");
```
[the remark this answers is not in the document]
Comment (hhund) [reply dsf-bpe/dsf-bpe-process-ping/src/test/java/org/highmed/dsf/bpe/start/Ping3MedicFromTtpExampleStarter.java]: fixed in 035048e

File: dsf-bpe/dsf-bpe-process-feasibility/src/main/java/org/highmed/dsf/bpe/FeasibilityProcessPluginDefinition.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+	@Override
+	public Stream<Class<?>> getSpringConfigClasses()
+	{
+		return Stream.of(FeasibilityConfig.class, FeasibilitySerializerConfig.class);
+	}
+
+	@Override
+	public ResourceProvider getResourceProvider(FhirContext fhirContext, ClassLoader classLoader)
+	{
+		var aCom = ActivityDefinitionResource.file("fhir/ActivityDefinition/computeSimpleFeasibility-0.3.0.xml");
```
[the remark this answers is not in the document]
Comment (hhund) [reply dsf-bpe/dsf-bpe-process-feasibility/src/main/java/org/highmed/dsf/bpe/FeasibilityProcessPluginDefinition.java]: done.

**Comment by hhund** [comment]
> I think there is a `plugin` folder with a README.md in `dsf-bpe-server-jetty` and a `processes` folder with a README.md in `dsf-bpe-server-jetty > docker` missing.

Added missing process folder in `dsf-bpe-server-jetty > docker`. `plugin` folder in `dsf-bpe-server-jetty` not needed, since starting the bpe server from within an IDE would ignore the contents anyways.

**Review by wetret — APPROVED** [review]

**Pull request merged**